link/perf_event.go

```diff
@@ -99,30 +99,31 @@ type perfEventLink struct {
 
 func (pl *perfEventLink) isLink() {}
 
-// Pinning requires the underlying perf event FD to stay open.
-//
-// | PerfEvent FD | BpfLink FD | Works |
-// |--------------|------------|-------|
-// | Open         | Open       | Yes   |
-// | Closed       | Open       | No    |
```

Comment:
So why is this OK now? Did something change wrt. this table? If I understand correctly, doing the following will not do the intuitive thing of "keep the attachment alive"?
```
link.Pin()
link.Close() // destroys the perf event fd
```

Maybe I misunderstood what "works" means in this context?

Reply:
so Pin/Close works for kprobe created via perf_link, because Pin increases the link ref, so it won't go away with the Close.. also the perf link increases ref of the kprobe perf event, so it stays when the original fd is closed

but it won't  work for kprobe via perf ioctl, which is fine

Reply:
> so Pin/Close works for kprobe created via perf_link, because Pin increases the link ref, so it won't go away with the Close.. also the perf link increases ref of the kprobe perf event, so it stays when the original fd is closed

But we still can't retrieve the FD and so can't load a pinned link?

> but it won't work for kprobe via perf ioctl, which is fine

How does it fail then? Silently?

Reply:
Ok, can we migrate the contents of comment to where it makes sense instead of outright deleting it? Probably on `perfEventIoctl` or similar.